### PR TITLE
QEMU exit logic + minimize crash fix

### PR DESCRIPTION
**Pull request by FuriousProgrammer**

### CODE REVIEW

File: ui/orbital.c [hunk 1]
```diff
@@ -535,24 +539,17 @@ static void* orbital_display_main(void* arg)
         ORBITAL_WIDTH, ORBITAL_HEIGHT, flags);
 
     count = 0;
-    if (!SDL_Vulkan_GetInstanceExtensions(ui.sdl_window, &count, NULL)) {
-        error_report("SDL_Vulkan_GetInstanceExtensions returned false");
-        return NULL;
-    }
+    SDL_Vulkan_GetInstanceExtensions(ui.sdl_window, &count, NULL);
```
Comment (AlexAltea) [inline ui/orbital.c]: What's the reasoning behind this change? `SDL_Vulkan_GetInstanceExtensions` can fail, and whether it does is returned in an `SDL_bool` value [1]. We should check it, and if it returns `SDL_FALSE` find+fix the root issue.

[1] https://wiki.libsdl.org/SDL_Vulkan_GetInstanceExtensions

File: ui/orbital.c [hunk 2]
```diff
@@ -535,24 +539,17 @@ static void* orbital_display_main(void* arg)
...
-    const char **extensionNames = (const char **)malloc((count + 3) * sizeof(char*));
+    const char **extensionNames = (const char **)malloc((count + 2) * sizeof(char*));
     extensionNames[count + 0] = VK_KHR_SURFACE_EXTENSION_NAME;
     extensionNames[count + 1] = VK_EXT_DEBUG_UTILS_EXTENSION_NAME;
-    extensionNames[count + 2] = VK_EXT_DEBUG_REPORT_EXTENSION_NAME;
-    if (!SDL_Vulkan_GetInstanceExtensions(ui.sdl_window, &count, &extensionNames[0])) {
-        error_report("SDL_Vulkan_GetInstanceExtensions returned false with extensions");
-        return NULL;
-    }
+    SDL_Vulkan_GetInstanceExtensions(ui.sdl_window, &count, &extensionNames[0]);
```
Comment (AlexAltea) [inline ui/orbital.c]: Same comment as above: https://github.com/AlexAltea/orbital-qemu/pull/19/files#r290837360

File: ui/orbital.c [hunk 3]
```diff
@@ -535,24 +539,17 @@ static void* orbital_display_main(void* arg)
...
+    SDL_Vulkan_GetInstanceExtensions(ui.sdl_window, &count, NULL);
     if (count == 0) {
         error_report("SDL_Vulkan_GetInstanceExtensions failed");
     }
 
-    const char **extensionNames = (const char **)malloc((count + 3) * sizeof(char*));
+    const char **extensionNames = (const char **)malloc((count + 2) * sizeof(char*));
     extensionNames[count + 0] = VK_KHR_SURFACE_EXTENSION_NAME;
     extensionNames[count + 1] = VK_EXT_DEBUG_UTILS_EXTENSION_NAME;
-    extensionNames[count + 2] = VK_EXT_DEBUG_REPORT_EXTENSION_NAME;
```
Comment (AlexAltea) [inline ui/orbital.c]: Why is this extension removed? It was added in https://github.com/AlexAltea/orbital-qemu/commit/fca99f2c5320333862639e2e8425cfca4483c19a.
If it causes issues in some systems it should be conditionally disabled.

**Comment by AlexAltea** [comment]
Thanks a lot for your patch! The QEMU-exiting part looks good to me, though I have some questions/concerns, about the other stuff (see above).

**Comment by FuriousProgrammer** [comment]
Uhh, I never made those changes (!) Think I might have pulled the wrong base on accident.

**Pull request merged**